### ISSUE TITLE
RateLimitApiException doesn't have response body content

### ISSUE THREAD

**Issue by fernandozpiccin** [auth0/auth0.net#479]
### Describe the problem you'd like to have solved

The RateLimitApiException doesn't contain the whole information given by the Auth0's Authentication API. The RateLimitApiException is thrown on every single HTTP 429 response, and, currently only contains the rate limit information - from the response's header. 
The Authentication API will also return HTTP 429 in other situations like when the Brute Force feature is enabled on tenant and an user has failed to login more than 10 times (Resource Owner Password flow). Please see the screenshot below
![image](https://user-images.githubusercontent.com/5048771/110139579-1a6b3a00-7dcb-11eb-9baf-783660e22b8c.png)
The "error" and "error_description" properties aren't available in RateLimitApiException.

### Describe the ideal solution

The RateLimitApiException should include data contained in response body - not only what is in the headers.

## Alternatives and current work-arounds

No alternatives have been found. The error handling factory cannot be replaced in the SDK

**Comment by frederikprijck** [comment]
Thanks for reporting.

I think this isn't a RateLimitException, so we should probably not handle the error as such in this case and only use RateLimitException when the headers are available or distinguish on anything other than the Status Code only.

Keep in mind using password grant is not recommended, as being explained here https://auth0.com/docs/flows/resource-owner-password-flow.

**Comment by fernandozpiccin** [comment]
Hi @frederikprijck - thanks for your reply.
Yes. We are aware of the limitations imposed by using the ROPG. Thanks for heads up.
The rate limits headers are also included in the response, therefore we cannot use that as factor to distinguish which exception should be created/thrown. It feels like the Auth API shouldn't be responding with a 429.
The implementation I did so far was to add the `ApiError` object as a member of `RateLimitApiException`. However, I just found out that `ApiError` doesn't contain a property for `error_description` (returned by the response shown above). It only contains a "message". I don't want to extend the ApiError object to include that as it might make things harder for other devs to consume due to many different variations of error description/error message.
Another alternative I've been considering was to make usage of the `Data` property, a dictionary, part of the base `Exception`. The only risk would be if we get a nested object as part of the response. I assume that wouldn't be the case as we aren't even considering the response body at all, at the moment.
Do you have any other ideas?

**Comment by fernandozpiccin** [comment]
I just noticed that there is a mapping for the json properties of `ApiError` in the `ApiErrorConverter` which makes all variations of error descriptions to be deserialized in the `message` property. That way, I've raised a PR for adding the `ApiError` as a property in the `RateLimitApiException`.
I haven't run the integration tests suite (see the reasoning in the PR). Although I integrated these changes into the project I'm working at the moment, and, now it does what I was expecting.
Happy to make any other suggested changes that will come from PR review.
Many thanks
